pkg/psalabelsyncer/podsecurity_label_sync_controller.go

```diff
@@ -250,57 +335,40 @@ func (c *PodSecurityAdmissionLabelSynchronizationController) syncNamespace(ctx c
 		return fmt.Errorf("unknown PSa level for namespace %q", ns.Name)
 	}
 
-	nsCopy := ns.DeepCopy()
-
-	var changed bool
-
-	if c.shouldEnforce {
-		if nsCopy.Labels[psapi.EnforceLevelLabel] != string(psaLevel) || nsCopy.Labels[psapi.EnforceVersionLabel] != currentPSaVersion {
-			changed = true
-			if nsCopy.Labels == nil {
-				nsCopy.Labels = map[string]string{}
-			}
-
-			nsCopy.Labels[psapi.EnforceLevelLabel] = string(psaLevel)
-			nsCopy.Labels[psapi.EnforceVersionLabel] = currentPSaVersion
+	managedNamespaces, err := extractNSFieldsPerManager(ns)
+	if err != nil {
+		if err != nil {
+			return fmt.Errorf("ns extraction failed: %w", err)
 		}
+	}
+	syncedLabels := enforcementLabels
+	if !c.shouldEnforce {
+		syncedLabels = loggingLabels
+	}
 
-		// cleanup audit and warn labels from version 4.11
-		// TODO: This can be removed in 4.13 and allow users set these as they wish
```

Comment:
Are we sure.  Aren't these used for our alerting?  Given the ability to yield to a different client, it seems like we'd want to continue setting these forever.

if we do not set these labels, then our default of "restricted" is used and this results in noise for namespaces that are not restricted because they constantly get warned and (I think), we get alerts that are not accurately indicating when a namespace will violate. 

In fact, is this leading to false positives today?

Reply:
The concerning flow is

1. our default is restricted for audit and warn
2. the namespace is set to privileged
3. we helpfully remove the audit and warn labels
4. a privileged pod is created and fails audit and warn
5. this causes our alert about how many future violates we would have to fail.

Reply:
I think you're right and we should set all.

My original thinking back then was we do not set these fields because we're only ever going to be setting enforcement globally, and we will no longer alert at that point. I still think that that's what we should do, however, enforcement can now be enabled via feature flags while the global config stays + since the labels can now be adjusted manually separately, setting all just seems more logical.

I'll fix that.